cql3/restrictions/statement_restrictions.hh

```diff
@@ -104,6 +104,8 @@ private:
 
     bool _has_queriable_regular_index = false, _has_queriable_pk_index = false, _has_queriable_ck_index = false;
 
+    std::optional<expr::expression> _where; ///< The entire WHERE clause.
```

Comment:
I like it the way it is.  Initializing it to `true` makes conjunct accumulation more complicated and doesn't distinguish between an absent WHERE clause and an explicit WHERE TRUE (when we start supporting that one day).

Reply:
The motivation is to make the code as regular as we can, with fewer special cases.

Reply:
> Why is accumulation more complicated? It's the same make_conjunction().

That would result in WHERE a>1 being represented as `true AND (a>1)`.  It's wrong and wasteful.

And since you're stateless, let me remind you that we already rejected the idea of `make_conjunction` optimizing TRUE out on the fly.

> The motivation is to make the code as regular as we can, with fewer special cases.

I think that's best achieved with the current code and would be undone by your suggestion.